components/src/store/index.ts

```diff
@@ -49,3 +49,40 @@ export const configureStore = () => {
 
     return store;
 }
+
+// normalizeState transforms slices of serialized state into a shape that conforms to
+// our current expectations.
+export function normalizeState(persistedState: any): Partial<AppState> {
```

Comment:
Nice! Maybe this will help out with the odd state issue we are hitting on *2pulumi pages

Reply:
I was thinking the same! We'll see. I'm somewhat surprised to see it not being set explicitly today, and TBH I haven't had time yet to investigate. But this should indeed ensure there's always at least something in that slot.